cmd/zfs/zfs_main.c

```diff
@@ -1105,6 +1111,11 @@ zfs_do_create(int argc, char **argv)
 		    "used when creating a volume\n"));
 		goto badusage;
 	}
+	if (nomount && type != ZFS_TYPE_FILESYSTEM) {
+		(void) fprintf(stderr, gettext("'-u' can only be "
+		    "used when creating a file system\n"));
```

Comment:
nit: It looks like all but a few of the error message use "filesystem" instead of "file system".